sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3328,13 +3328,25 @@ object functions {
   }
 
   /**
-   * Sorts the input array in ascending order. The elements of the input array must be orderable.
-   * Null elements will be placed at the end of the returned array.
+   * Sorts the input array for the given column in ascending order,
+   * according to the natural ordering of the array elements.
+   * Null elements will be placed at the beginning of the returned array.
    *
    * @group collection_funcs
    * @since 2.4.0
    */
-  def array_sort(e: Column): Column = withExpr { ArraySort(e.expr) }
+  def array_sort(e: Column): Column = array_sort(e, asc = true)
+
+  /**
+    * Sorts the input array for the given column in ascending or descending order,
+    * according to the natural ordering of the array elements.
+    * Null elements will be placed at the beginning of the returned array in ascending order or
+    * at the end of the returned array in descending order.
+    *
+    * @group collection_funcs
+    * @since 2.4.0
+    */
+  def array_sort(e: Column, asc: Boolean): Column = withExpr { ArraySort(e.expr, lit(asc).expr) }
```

Comment:
asc: Boolean -> asc: Column (see comments on the top of `functions.scala`)

Reply:
So asc should be a Column?

Are you referring to this comment? 

` * This function APIs usually have methods with `Column` signature only because it can support not
 * only `Column` but also other types such as a native string. The other variants currently exist
 * for historical reasons.`